hack/install.sh

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+DEFAULT_BIN_DIR="/usr/local/bin"
+DEFAULT_CLI="karmadactl"
+CLI=${1:-"${DEFAULT_CLI}"}
+BIN_DIR=${2:-"${DEFAULT_BIN_DIR}"}
```

Comment:
```bash
# INSTALL_CLI_TYPE supports 'karmadactl' or 'kubectl-karmada'.
INSTALL_CLI_TYPE=${INSTALL_CLI_TYPE:-"karmadactl"}
# INSTALL_CLI_VERSION represents which version should be installed.
# Defaults to 'latest' means latest release version.
INSTALL_CLI_VERSION=${INSTALL_CLI_VERSION:-"latest"}
# INSTALL_LOCATION represents where to put the CLI.
INSTALL_LOCATION=${INSTALL_LOCATION:-"/usr/local/bin"}
```

We can put all variables that support customized at the head of the script.

In addition, we'd better describe the usage at the head of the script.